CITATION.cff

```diff
@@ -0,0 +1,42 @@
+cff-version: 1.2.0
+title: >-
+  TapeAgents: a Holistic Framework for Agent Development and
+  Optimization
+message: 'If you use this software, please cite it as below.'
```

Comment:
Do we need to enforce citations for anyone using the repo? It looks like overkill to me, I would expect it to be like "If you want to refer to our paper or code, please cite ..."